<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Differential expression analysis: Correlation of 4 subgroups</title>
  <style>
    html {
      line-height: 1.5;
      font-family: Georgia, serif;
      font-size: 20px;
      color: #1a1a1a;
      background-color: #fdfdfd;
    }
    body {
      margin: 0 auto;
      max-width: 36em;
      padding-left: 50px;
      padding-right: 50px;
      padding-top: 50px;
      padding-bottom: 50px;
      hyphens: auto;
      word-wrap: break-word;
      text-rendering: optimizeLegibility;
      font-kerning: normal;
    }
    @media (max-width: 600px) {
      body {
        font-size: 0.9em;
        padding: 1em;
      }
    }
    @media print {
      body {
        background-color: transparent;
        color: black;
        font-size: 12pt;
      }
      p, h2, h3 {
        orphans: 3;
        widows: 3;
      }
      h2, h3, h4 {
        page-break-after: avoid;
      }
    }
    p {
      margin: 1em 0;
    }
    a {
      color: #1a1a1a;
    }
    a:visited {
      color: #1a1a1a;
    }
    img {
      max-width: 100%;
    }
    h1, h2, h3, h4, h5, h6 {
      margin-top: 1.4em;
    }
    h5, h6 {
      font-size: 1em;
      font-style: italic;
    }
    h6 {
      font-weight: normal;
    }
    ol, ul {
      padding-left: 1.7em;
      margin-top: 1em;
    }
    li > ol, li > ul {
      margin-top: 0;
    }
    blockquote {
      margin: 1em 0 1em 1.7em;
      padding-left: 1em;
      border-left: 2px solid #e6e6e6;
      color: #606060;
    }
    code {
      font-family: Menlo, Monaco, 'Lucida Console', Consolas, monospace;
      font-size: 85%;
      margin: 0;
    }
    pre {
      margin: 1em 0;
      overflow: auto;
    }
    pre code {
      padding: 0;
      overflow: visible;
    }
    .sourceCode {
     background-color: transparent;
     overflow: visible;
    }
    hr {
      background-color: #1a1a1a;
      border: none;
      height: 1px;
      margin: 1em 0;
    }
    table {
      margin: 1em 0;
      border-collapse: collapse;
      width: 100%;
      overflow-x: auto;
      display: block;
      font-variant-numeric: lining-nums tabular-nums;
    }
    table caption {
      margin-bottom: 0.75em;
    }
    tbody {
      margin-top: 0.5em;
      border-top: 1px solid #1a1a1a;
      border-bottom: 1px solid #1a1a1a;
    }
    th {
      border-top: 1px solid #1a1a1a;
      padding: 0.25em 0.5em 0.25em 0.5em;
    }
    td {
      padding: 0.125em 0.5em 0.25em 0.5em;
    }
    header {
      margin-bottom: 4em;
      text-align: center;
    }
    #TOC li {
      list-style: none;
    }
    #TOC a:not(:hover) {
      text-decoration: none;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Differential expression analysis: Correlation of 4 subgroups</h1>
</header>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(curl)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using libcurl 7.84.0 with LibreSSL/3.3.6</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter()     masks stats::filter()
✖ dplyr::lag()        masks stats::lag()
✖ readr::parse_date() masks curl::parse_date()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;quarto&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>protein_data <span class="ot">&lt;-</span> <span class="fu">read_csv2</span>(<span class="st">&quot;../_raw/77_cancer_proteomes_CPTAC_itraq.csv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ Using &quot;&#39;,&#39;&quot; as decimal and &quot;&#39;.&#39;&quot; as grouping mark. Use `read_delim()` for more control.
New names:Rows: 12553 Columns: 87── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;;&quot;
chr  (3): RefSeq_accession_number, gene_symbol, gene_name
dbl (84): ...1, AO-A12D.01TCGA, C8-A131.01TCGA, AO-A12B.01TCGA, BH-A18Q.02TC...
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb8" class="sourceCode cell-code">
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;../_raw/clinical_data_breast_cancer.csv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 105 Columns: 30
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (17): Complete TCGA ID, Gender, ER Status, PR Status, HER2 Final Status,...
dbl (13): Age at Initial Pathologic Diagnosis, Days to Date of Last Contact,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb10" class="sourceCode cell-code">
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>PAM50_file <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;../_raw/PAM50_proteins.csv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 100 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (4): GeneSymbol, RefSeqProteinID, Species, Gene Name

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div id="cb12" class="sourceCode cell-code">
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#write_csv2 converts the count matrix to a variable of characters</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv2</span>(protein_data, <span class="st">&quot;../data/protein_load.csv&quot;</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv2</span>(clinical_data, <span class="st">&quot;../data/clinical_load.csv&quot;</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv2</span>(PAM50_file, <span class="st">&quot;../data/PAM50_load.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<h2 id="toc-title">Table of contents</h2>
<ul>
<li><a href="#cleaning-of-data" id="toc-cleaning-of-data" class="nav-link active">Cleaning of data</a></li>
</ul>
</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<section id="cleaning-of-data" class="level2">
<h2 class="anchored" data-anchor-id="cleaning-of-data" id="cleaning-of-data-1">Cleaning of data</h2>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ purrr     1.0.2
✔ forcats   1.0.0     ✔ tibble    3.2.1
✔ ggplot2   3.4.3     ✔ tidyr     1.3.0
✔ lubridate 1.9.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>protein_data <span class="ot">&lt;-</span> <span class="fu">read_csv2</span>(<span class="st">&quot;../data/protein_load.csv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ Using &quot;&#39;,&#39;&quot; as decimal and &quot;&#39;.&#39;&quot; as grouping mark. Use `read_delim()` for more control.
Rows: 12553 Columns: 87── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;;&quot;
chr  (3): RefSeq_accession_number, gene_symbol, gene_name
dbl (84): ...1, AO-A12D.01TCGA, C8-A131.01TCGA, AO-A12B.01TCGA, BH-A18Q.02TC...
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;../data/clinical_load.csv&quot;</span>) </span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 105 Columns: 30
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;;&quot;
chr (17): Complete TCGA ID, Gender, ER Status, PR Status, HER2 Final Status,...
dbl (13): Age at Initial Pathologic Diagnosis, Days to Date of Last Contact,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>PAM50_file <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;../data/PAM50_load.csv&quot;</span>) </span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 100 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot; &quot;
chr (2): GeneSymbol;RefSeqProteinID;Species;Gene, Name

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div id="cb10" class="sourceCode cell-code">
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Removal of negative controls</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> protein_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(RefSeq_accession_number,gene_symbol,gene_name)) <span class="sc">|</span><span class="er">&gt;</span>  </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Remove columns with the names: &quot;RefSeq_accession_number,gene_symbo</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>          <span class="co"># ,gene_name&quot;</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Returns all colnames in a list</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="at">pattern =</span> <span class="fu">regex</span>(<span class="st">&quot;TCGA$&quot;</span>))</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Removes all columns that does not contain TCGA as their last letters. THis is because the dataset have 3 controls, </span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>          <span class="co"># named: &quot;263d3f-I.CPTAC&quot; &quot;blcdb9-I.CPTAC&quot; &quot;c4155b-C.CPTAC&quot;</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Creation of the count_matrix</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>count_matrix_1 <span class="ot">&lt;-</span> protein_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(RefSeq_accession_number, <span class="fu">all_of</span>(samples)) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Creating a subset with  RefSeq_accession_number and all samples,</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>          <span class="co"># excluding the control samples</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>)))<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>          <span class="co"># replace all NA values in the count-matrix with 0</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;RefSeq_accession_number&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Transfroms the dataset to a pivot longer with sample ID as names and</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>          <span class="co"># values as expression counts</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">str_extract</span>(name, <span class="st">&quot;[^.]+&quot;</span>),</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">name =</span> <span class="fu">str_replace</span>(name, <span class="st">&quot;-&quot;</span>, <span class="st">&quot;_&quot;</span>),</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">name =</span> <span class="fu">str_c</span>(<span class="st">&quot;TCGA&quot;</span>, name, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>           <span class="co"># Removing everyting after the first &quot;.&quot; in the name column. </span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>           <span class="co"># &quot;C8-A130.02TCGA&quot; becomes &quot;C8-A130&quot;, a &quot;-&quot; becomes a &quot;_&quot;</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> name, <span class="at">values_from =</span> value, <span class="at">values_fn =</span> {list}) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>         <span class="co"># Transforms the dataset back to a pivot wider with columns as sample ID </span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>         <span class="co"># and values as expression counts</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(<span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;RefSeq_accession_number&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>        <span class="co"># We unnest the dataset</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(RefSeq_accession_number,<span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">id =</span> <span class="st">&quot;RefSeq_accession_number&quot;</span>)</span></code></pre></div>
</div>
</div>
<p>Define list of patients that are in the count matrix (we have proteomics data from)</p>
<div class="cell">
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>list_of_patients <span class="ot">&lt;-</span> count_matrix_1 <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="at">pattern =</span> <span class="fu">regex</span>(<span class="st">&quot;TCGA&quot;</span>))</span></code></pre></div>
</div>
</div>
<p>Renaming the clincal data set and filtering for only the patent ID’s that are in the count matrix.</p>
<div class="cell">
<div id="cb12" class="sourceCode cell-code">
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>clinical_data_final <span class="ot">&lt;-</span> clinical_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">id =</span> <span class="st">&quot;Complete TCGA ID&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>          <span class="co"># renames &quot;Complete TCGA ID&quot; to id</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">str_replace_all</span>(id, <span class="st">&quot;-&quot;</span>, <span class="st">&quot;_&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>          <span class="co"># mutates the ids to having to -, but only _</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename_with</span>( <span class="sc">~</span> <span class="fu">str_to_lower</span>(<span class="fu">str_replace_all</span>(.x, <span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>))) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pam50 =</span> pam50_mrna)<span class="sc">|</span><span class="er">&gt;</span> <span class="co">#rename for the same of ease later on</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(id <span class="sc">%in%</span> <span class="fu">colnames</span>(count_matrix_1))<span class="co"># include only Id that are in our count matrix</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>clinical_data_final <span class="ot">&lt;-</span> clinical_data_final <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(id <span class="sc">%in%</span> list_of_patients)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb13" class="sourceCode cell-code">
<div class="sourceCode" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(clinical_data_final, <span class="st">&quot;../data/clinical_data.tsv&quot;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(count_matrix_1, <span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PAM50_file, <span class="st">&quot;../data/PAM50_file.tsv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb14" class="sourceCode cell-code">
<div class="sourceCode" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</section>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<h2 id="toc-title">Table of contents</h2>
<ul>
<li><a href="#making-the-full-data-set" id="toc-making-the-full-data-set" class="nav-link active">Making the full data set:</a></li>
</ul>
</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<section id="making-the-full-data-set" class="anchored level3" data-anchor-id="making-the-full-data-set">
<h3 class="anchored" data-anchor-id="making-the-full-data-set">Making the full data set:</h3>
<p>First, the count matrix is transposed using <code>pivor_longer</code> and <code>pivot_wider</code>. This is done to get the subjects as rows and proteins as columns.</p>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/clinical_data.tsv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>protein_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>, <span class="at">header =</span> T)</span></code></pre></div>
</div>
</div>
<p>Now, the clinical data is loaded and joined to the protein data in a dataset called <code>full_data</code>. Additionally, new columns are made, storing binary information of which subgroup (pam50-classification) each person is in.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Read the clinial data</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>protein_data <span class="ot">&lt;-</span> protein_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Load the count matrix</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>id, <span class="at">names_to =</span> <span class="st">&#39;variable&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;value&#39;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Pivot longer. First step in transposing the matrix. </span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> id, <span class="at">values_from =</span> value) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Pivot wider - second step in transposing the matrix</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">id =</span> variable)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Rename the column &#39;variable&#39; to &#39;id&#39;</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> clinical_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>( </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGEA_HER2 =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>      pam50 <span class="sc">==</span> <span class="st">&quot;HER2-enriched&quot;</span> <span class="sc">~</span> <span class="dv">1</span>, </span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="dv">0</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>    ), <span class="co"># ^ Make a new column called &#39;DGEA_HER2&#39; storing binary information of </span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>       <span class="co"># whether the individual is positive (1) or negative (0) for HER2</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGEA_Luminal_A =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>      pam50 <span class="sc">==</span> <span class="st">&quot;Luminal A&quot;</span> <span class="sc">~</span> <span class="dv">1</span>, </span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="dv">0</span></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  ),   <span class="co"># ^ Make a new column called &#39;DGEA_Luminal_A&#39; storing binary information of </span></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>       <span class="co"># whether the individual is positive (1) or negative (0) for Luminal A</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGEA_Luminal_B =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>      pam50 <span class="sc">==</span> <span class="st">&quot;Luminal B&quot;</span> <span class="sc">~</span> <span class="dv">1</span>, </span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="dv">0</span></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>      ),<span class="co"># ^ Make a new column called &#39;DGEA_Luminal_B&#39; storing binary information of </span></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>       <span class="co"># whether the individual is positive (1) or negative (0) for Luminal B</span></span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">DGEA_Basal_like =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>      pam50 <span class="sc">==</span> <span class="st">&quot;Basal-like&quot;</span> <span class="sc">~</span> <span class="dv">1</span>, </span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="dv">0</span>),</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">er_status =</span> <span class="fu">case_when</span>(er_status <span class="sc">==</span> <span class="st">&quot;Negative&quot;</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>                        er_status <span class="sc">==</span> <span class="st">&quot;Positive&quot;</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span>),</span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">pr_status =</span> <span class="fu">case_when</span>(pr_status <span class="sc">==</span> <span class="st">&quot;Negative&quot;</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>                        pr_status <span class="sc">==</span> <span class="st">&quot;Positive&quot;</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span>),</span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>  <span class="at">her2_final_status =</span> <span class="fu">case_when</span>(her2_final_status <span class="sc">==</span> <span class="st">&quot;Negative&quot;</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>                        her2_final_status <span class="sc">==</span> <span class="st">&quot;Positive&quot;</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>                        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span>)) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_data, <span class="at">by =</span> <span class="st">&quot;id&quot;</span>)</span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Leftjoin the protein expression data to make a big dataframe with everything</span></span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(full_data, <span class="st">&quot;../data/full_data.tsv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</section>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<h2 id="toc-title">Table of contents</h2>
<ul>
<li><a href="#introduction-to-the-data" id="toc-introduction-to-the-data" class="nav-link active">Introduction to the data</a>
<ul>
<li><a href="#project-aim" id="toc-project-aim" class="nav-link">Project aim</a></li>
</ul></li>
</ul>
</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<section id="introduction-to-the-data" class="level1">
<h1>Introduction to the data</h1>
<p>The data is derived from a proteogenomic study investigating how DNA mutations affect the expression of the proteins in breast cancer tissue in patients. When DNA is changing it has an impact on the proteome i.e., the functional molecular part of the organisms. In this context the proteins are responsible for enzymatic reactions / signalling, and they are controlling cell division, and DNA repair, etc.</p>
<p>The dataset contains information on TCGA ID (Clinical Proteomic Tumour Analysis Consortium; NCI/NIH), RefSeq_accession_number (RefSeq database), and from the Uniprot database following variables were available: Entry name, Protein name, Gene symbol, Gene name, Organism (Homo Sapiens), and Length. Finally, datasets contain information on breast cancer TNM classification (AJCC), tumour receptor status for oestrogen, progesterone, and HER2 - incl conversion of receptor status (influencing treatment effect and survival), survival, and, importantly for our focus in the following project, PAM50 (Prediction Analysis of Microarray 50) classification.</p>
<p>PAM50 classification is based on a gene signature of 50 genes considered representative of different biological pathways and molecular features. It is used in breast-cancer research to categorize tumors based on microarray or RNA-sequencing data. The four classifications are:</p>
<ul>
<li><p>Luminal A</p></li>
<li><p>Luminal B</p></li>
<li><p>HER2-enriched</p></li>
<li><p>Basal-like</p></li>
</ul>
<p>In the present study, iTRAQ proteome profiling was performed on 77 breast cancer samples each generally expressing around 12.000 proteins. The protein data was divided by use of K-means clustering into groups with different breast cancer subtypes expressing unique protein signatures.</p>
<section id="project-aim" class="anchored level2" data-anchor-id="project-aim">
<h2 class="anchored" data-anchor-id="project-aim">Project aim</h2>
<p>In this project, we wanted to see how well the proteome profiling relates to the PAM50 classification.</p>
<p><br />
To show the distribution of how the 77 samples in the data set were classified by the PAM50 a simple plot was made.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
</div>
<p>A barplot was chosen for visualisation:</p>
<div class="cell">
<div id="cb2" class="sourceCode cell-code">
<div class="sourceCode" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>barplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> clinical_data, </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> pam50)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> pam50)) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Distribution of patients in the data set based on PAM50 classification&quot;</span>,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;PAM50 categories&quot;</span>,</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;No. of patients&quot;</span>) <span class="sc">+</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">19.25</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>)<span class="co"># horisontal line to show an equal distribution</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>barplot</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="04_describe_DONE_files/figure-html/Analysis-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/pam50distribution.png&quot;</span>, barplot, <span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<p>From the barplot it is evident, that the patients are fairly evenly distributed with the HER2-enriched category being under-represented and both the Luminal A and B slightly over what would be completely even distribution.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</section>
</section>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<h2 id="toc-title">Table of contents</h2>
<ul>
<li><a href="#analysis-based-on-entire-proteomics-data-set" id="toc-analysis-based-on-entire-proteomics-data-set" class="nav-link active">Analysis based on entire proteomics data set</a></li>
</ul>
</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<p>To investigate if the clinically assigned pam50 categories display in the proteomics dataset, we also wanted to do a hierachial clustering analysis. This was visualised using dendrograms.</p>
<p>First, packages and data are loaded.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  id = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
</div>
<section id="analysis-based-on-entire-proteomics-data-set" class="anchored level3" data-anchor-id="analysis-based-on-entire-proteomics-data-set">
<h3 class="anchored" data-anchor-id="analysis-based-on-entire-proteomics-data-set">Analysis based on entire proteomics data set</h3>
<p>Here, the hierachial clustering is done and plotted as a dendrogram using the <code>hc_dendro_plot</code> function from the <code>99_proj_func.R</code> file. The function first takes the count matrix and calculates the hierarchical clustering. Hereafter, it plots the results as a dendrogram while colouring for pam50-subtypes</p>
<div class="cell">
<div id="cb2" class="sourceCode cell-code">
<div class="sourceCode" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>ddg_plot <span class="ot">&lt;-</span> <span class="fu">hc_dendro_plot</span>(<span class="at">count_matrix =</span> cm, <span class="at">data_type =</span> <span class="st">&quot;full proteomics dataset&quot;</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">colour =</span> <span class="st">&quot;pam50&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>ddg_plot</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="05_analysis_clustering_DONE_files/figure-html/Hierachial%20clustering-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save dendrogram plot</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/hc_dendro_all.png&quot;</span>, ddg_plot, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<p><strong>Conclusion</strong>: From the colored labels, it is apparent, that the samples don’t group by pam50 categories when a hierachial clustering is made on the entire proteomics dataset.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</section>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<p>It was investigated if the data set would show any clustering resembling that of the PAM50 criteria, based on the entire proteomics data set. First a PCA analysis was conducted.</p>
<p>First, required packages were loaded.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors

Attaching package: &#39;cowplot&#39;


The following object is masked from &#39;package:lubridate&#39;:

    stamp</code></pre>
</div>
</div>
<p>Next, the required data was loaded. For this analysis it was the count matrix, which was transformed (rows to columns and vice versa), and the clinical data.</p>
<div class="cell">
<div id="cb2" class="sourceCode cell-code">
<div class="sourceCode" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>, <span class="at">header =</span> T) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>id, <span class="at">names_to =</span> <span class="st">&#39;variable&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;value&#39;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> id, <span class="at">values_from =</span> value) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">id =</span> variable) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&#39;id&#39;</span>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/clinical_data.tsv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header =</span> T)</span></code></pre></div>
</div>
</div>
<p>Next, the PCA was done by first scaling the count matrix, and then the doing the actual principal component analysis using the prcomp() function. From the pca output, information about the variance-explanation by each principel component was extracted and stored in the pca_results using the tidy() function.</p>
<div class="cell">
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> cm <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  stats<span class="sc">::</span><span class="fu">prcomp</span>(<span class="at">center=</span><span class="cn">TRUE</span>, <span class="at">scale=</span><span class="cn">TRUE</span>) <span class="co"># transform data to principal components</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract variance explained by the PCs</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>pca_results <span class="ot">&lt;-</span> pca <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="st">&quot;pcs&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Using broom::tidy to extract results</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">pluck</span>(<span class="st">&quot;percent&quot;</span>) <span class="co"># extracting the explained varaince of each PC</span></span></code></pre></div>
</div>
</div>
<p>These variables, were used for creating a visual presentation of the principal component analysis, in the form of a classic PCA plot of the first to principal components. Here the individual samples were coloured based on their pam50 classification.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create PCA plot</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>pca_plot <span class="ot">&lt;-</span> pca <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pluck</span>(<span class="st">&quot;x&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC1, </span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">y=</span>PC2, </span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> clinical_data <span class="sc">|</span><span class="er">&gt;</span> dplyr<span class="sc">::</span><span class="fu">pull</span>(<span class="st">&quot;pam50&quot;</span>), <span class="co"># color by pam50 from clinical_data</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">fill =</span> clinical_data <span class="sc">|</span><span class="er">&gt;</span> dplyr<span class="sc">::</span><span class="fu">pull</span>(<span class="st">&quot;pam50&quot;</span>))) <span class="sc">+</span></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>,</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>, </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span> <span class="co"># font size of text</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>   <span class="fu">background_grid</span>() <span class="sc">+</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="fu">str_c</span>(<span class="st">&#39;PC1,  &#39;</span>, <span class="fu">round</span>(pca_results[<span class="dv">1</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">1</span>), <span class="st">&#39;% VE&#39;</span>)) <span class="sc">+</span> </span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ylab</span>(<span class="fu">str_c</span>(<span class="st">&#39;PC2,  &#39;</span>, <span class="fu">round</span>(pca_results[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">1</span>), <span class="st">&#39;% VE&#39;</span>)) <span class="sc">+</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>   <span class="fu">labs</span>(</span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">title =</span> <span class="st">&quot;PCA on entire proteomics dataset&quot;</span>,</span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">color =</span> <span class="st">&quot;pam50&quot;</span>,</span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">fill =</span> <span class="st">&quot;pam50&quot;</span></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a><span class="co">#print the plot</span></span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>pca_plot</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="06_analysis_PCA_cluster_DONE_files/figure-html/PCA%20plot-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save the plot</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/PCA1.png&quot;</span>, pca_plot, <span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<p>In this PCA based on all the proteomics data, we see no obvious clustering.</p>
<div class="cell">
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/full_data.tsv&quot;</span>, <span class="at">header =</span> T)</span></code></pre></div>
</div>
</div>
<p>In the following, the actual gene expression analysis takes place. 4 analyses are made - one for each subtype of PAM50. Each subtype is compared to all the rest of the data.</p>
<p>The function for making the linear predictions are made in the <code>99_proj_func.R</code> file and are called to this file using <code>source()</code>.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Her 2</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">linear_model_deg</span>(<span class="at">data =</span> full_data, <span class="at">test =</span> <span class="st">&quot;DGEA_HER2&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using an external vector in selections was deprecated in tidyselect 1.1.0.
ℹ Please use `all_of()` or `any_of()` instead.
  # Was:
  data %&gt;% select(test)

  # Now:
  data %&gt;% select(all_of(test))

See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.</code></pre>
</div>
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Luminal A</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>Luminal_A <span class="ot">&lt;-</span> <span class="fu">linear_model_deg</span>(<span class="at">data =</span> full_data,<span class="at">test =</span> <span class="st">&quot;DGEA_Luminal_A&quot;</span>)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Luminal B</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>Luminal_B <span class="ot">&lt;-</span> <span class="fu">linear_model_deg</span>(<span class="at">data =</span> full_data,<span class="at">test =</span> <span class="st">&quot;DGEA_Luminal_B&quot;</span>)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Basal like</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>Basal_like <span class="ot">&lt;-</span> <span class="fu">linear_model_deg</span>(<span class="at">data =</span> full_data,<span class="at">test =</span> <span class="st">&quot;DGEA_Basal_like&quot;</span>)</span></code></pre></div>
</div>
</div>
<p>Save the results to the data folder:</p>
<div class="cell">
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the results for easy import into next steps of the analysis.</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(HER2, <span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(Luminal_A, <span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(Luminal_B, <span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(Basal_like, <span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb8" class="sourceCode cell-code">
<div class="sourceCode" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co">#quarto::quarto_render(&quot;07_analysis_diff_exp.qmd&quot;,output_format = &quot;html&quot;,execute = T)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="co">#file.rename(from = &quot;07_analysis_diff_exp.html&quot;,</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co">#            to = &quot;../results/07_analysis_diff_exp.html&quot;)</span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<p>Here, we want to see if we can determine the PAM50 groups using only the significant proteins from the linear models.</p>
<p>First, we make a PCA using the</p>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;../data/clinical_data.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 77 Columns: 30
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (17): id, gender, er_status, pr_status, her2_final_status, tumor, tumor-...
dbl (13): age_at_initial_pathologic_diagnosis, days_to_date_of_last_contact,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing HER2 condition based on count matrix</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>lumA <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing luminal A condition based on count matrix</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>lumB <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing Luminal B condition based on count matrix</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>basal_like <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing Basal like condition based on count matrix</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>protein_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co"># loading the count matrix</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots <span class="ot">&lt;-</span> <span class="fu">c</span>(HER2 <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein),</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>                  lumA <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>                  lumB <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>                  basal_like <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein)) <span class="co"># Extracting only significant hits from differential expression analysis. Significant hits are defined at genes that have an adjusted P-value under 0.05. </span></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots_unique <span class="ot">&lt;-</span> <span class="fu">unique</span>(signif_fdr_prots) <span class="co"># extracting only unique gene names from significant hits </span></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a><span class="co">#length(signif_fdr_prots_unique)</span></span></code></pre></div>
</div>
</div>
<p>We can now isolate only significant hits form our count matrix.</p>
<div class="cell">
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>count_matrix_reduced <span class="ot">&lt;-</span> protein_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(id <span class="sc">%in%</span> signif_fdr_prots_unique) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Filtering so only differential expressed genes are present. </span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>id, <span class="at">names_to =</span> <span class="st">&quot;id2&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Transforming to pivot_longer</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> id, <span class="at">values_from =</span> value) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># transforming it again to pivot_wider. to have the correct structure - gene names as columns and samples as rows</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;id2&quot;</span>)</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>count_matrix_reduced <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;id&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">write_tsv</span>(<span class="st">&quot;../data/count_matrix_significant_proteins.tsv&quot;</span>) <span class="co"># saving results as a tsv file. </span></span></code></pre></div>
</div>
</div>
<p>PCA analysis of count matrix with only differentially expressed genes.</p>
<div class="cell">
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> count_matrix_reduced <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prcomp</span>(<span class="at">center =</span> <span class="cn">TRUE</span>,</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="cn">TRUE</span>) <span class="co"># prcomp is a function in R to transform the data into principal components. </span></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Scree</span></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>pca  <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="st">&quot;pcs&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span>  <span class="co"># From broom, we&#39;re using the tidy function to extract the nested list &quot;pcs&quot;</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> percent <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">|</span><span class="er">&gt;</span>  <span class="co"># Overriding the column Percent by multiplying current value with 100</span></span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, </span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> percent)) <span class="sc">+</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="co"># geom_Col function creates a barplot from the global values X as the princial compoennts and &quot;Y&quot; as the results form Principal components.</span></span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> <span class="co"># base_size = font size of the text </span></span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Scree Plot of PCA of significant proteins&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="08_analysis_kmeans_DONE_files/figure-html/analysis%202-1.png" class="img-fluid" width="672" /></p>
</div>
</div>
<div class="cell">
<div id="cb8" class="sourceCode cell-code">
<div class="sourceCode" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>pca_plot_axes_labels <span class="ot">&lt;-</span> pca <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="st">&quot;eigenvalues&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># extracting tidy function from broom package</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lbl =</span> <span class="fu">str_c</span>(<span class="st">&quot;PC&quot;</span>, PC, <span class="st">&quot;, VE = &quot;</span>, <span class="fu">round</span>(percent<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>), <span class="st">&quot;%&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Creating a new variable called lbl where values are created by concatenating the principal compoennts and its respective variance into a character string</span></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="st">&quot;lbl&quot;</span>) <span class="co"># pulling the lbl value out</span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a> <span class="co"># Loading the clinical data.</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>pca_aug <span class="ot">&lt;-</span> pca <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">augment</span>(count_matrix_reduced)<span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Using the augment function from broom to extend the pca dataset with our count matrix of differentially expressed genes</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">id =</span> .rownames) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># renaming the .rownames column to id</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clinical_data <span class="sc">|</span><span class="er">&gt;</span>  <span class="fu">select</span>(<span class="fu">c</span>(id, pam50)), <span class="co"># appending the pam50 columns from the clinical dataset to the pca dataset, based on id. </span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">&quot;id&quot;</span>)</span></code></pre></div>
</div>
</div>
<p>Now, we run the k-means algorithm using n = 4, as we have 4 groups we want to determine:</p>
<div class="cell">
<div id="cb10" class="sourceCode cell-code">
<div class="sourceCode" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>k_means_clusters <span class="ot">&lt;-</span> pca_aug <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(.fittedPC1, .fittedPC2)) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># selecting PC1 and PC2</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kmeans</span>(<span class="at">centers =</span> <span class="dv">4</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># initializing Kmeans clustering with 4 clusters</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pluck</span>(<span class="st">&quot;cluster&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># extracting &quot;cluster&quot; list.</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.factor</span>() <span class="co"># converting the results to a factor for coloring and filling of PCA plot. </span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>PCA_kmeans <span class="ot">&lt;-</span> pca_aug <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> k_means_clusters) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> .fittedPC1, <span class="co"># PC1 on x axis</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> .fittedPC2, <span class="co"># PC2 on y axis</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="fu">factor</span>(k_means_clusters), <span class="co"># color based on pam50 </span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> <span class="fu">factor</span>(k_means_clusters))) <span class="sc">+</span> <span class="co"># fill based on pam50</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>) <span class="sc">+</span><span class="co"># creating a vertical line that interceps at x = 0</span></span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span><span class="co"># creating a horizontal line that intercepts at y = 0</span></span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">6</span>, </span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="co"># transparency of the dots is set to 0.5</span></span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">shape =</span> pam50)) <span class="sc">+</span></span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a>          <span class="co"># global are already given above and we not just apply design for the dots </span></span>
<span id="cb65-14"><a href="#cb65-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb65-15"><a href="#cb65-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;PCA categorized through clusters&quot;</span>, <span class="co"># title of the data</span></span>
<span id="cb65-16"><a href="#cb65-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">pluck</span>(pca_plot_axes_labels, <span class="dv">1</span>), <span class="co">#name of x axis is extracted through the first value of pca_plot_axes_labels variable</span></span>
<span id="cb65-17"><a href="#cb65-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">pluck</span>(pca_plot_axes_labels, <span class="dv">2</span>), <span class="co">#name of y axis is extracted through the second value of pca_plot_axes_labels variable</span></span>
<span id="cb65-18"><a href="#cb65-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Cluster factor&quot;</span>,</span>
<span id="cb65-19"><a href="#cb65-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;Cluster factor&quot;</span></span>
<span id="cb65-20"><a href="#cb65-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb65-21"><a href="#cb65-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="co">#setting theme to bw</span></span>
<span id="cb65-22"><a href="#cb65-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">fill =</span> pam50, <span class="at">color =</span> pam50), <span class="co">#adding informtion on the actual pam50 classifications</span></span>
<span id="cb65-23"><a href="#cb65-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">type =</span> <span class="st">&quot;t&quot;</span>, </span>
<span id="cb65-24"><a href="#cb65-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, </span>
<span id="cb65-25"><a href="#cb65-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.1</span>) </span>
<span id="cb65-26"><a href="#cb65-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-27"><a href="#cb65-27" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(PCA_kmeans)</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="08_analysis_kmeans_DONE_files/figure-html/save%20results-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb12" class="sourceCode cell-code">
<div class="sourceCode" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/PCA_with_clusters.png&quot;</span>, PCA_kmeans, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb13" class="sourceCode cell-code">
<div class="sourceCode" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="co">#read files.</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>lumA <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>lumB <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>basal_like <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
</div>
<p>To see dispersion of differentially expressed genes, we visualize then through a volcanoplot. to make the volcanoplot, we call the function <code>volcano_plot()</code> from the <code>99_proj_func.R</code> file.</p>
<div class="cell">
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> HER2, <span class="at">subtype =</span> <span class="st">&quot;HER2&quot;</span>)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> lumA, <span class="at">subtype =</span> <span class="st">&quot;Luminal A&quot;</span>)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> lumB, <span class="at">subtype =</span> <span class="st">&quot;Luminal A&quot;</span>)</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> basal_like, <span class="at">subtype =</span> <span class="st">&quot;Basal like&quot;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb12" class="sourceCode cell-code">
<div class="sourceCode" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co">#loads patchwork library to combine the four volcanoplots where / creates to row #and two columns, and collect ensures that legends are shared.</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="co">#the file with combined plot is saved as png file in &quot;/results/ directory with specified dimentions.</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>volcano_plots <span class="ot">&lt;-</span> (plot1 <span class="sc">+</span> plot2)<span class="sc">/</span>(plot3 <span class="sc">+</span> plot4) <span class="sc">+</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/volcano_plots.png&quot;</span>, volcano_plots, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">width =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 14 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 141 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
</div>
<div class="cell">
<div id="cb15" class="sourceCode cell-code">
<div class="sourceCode" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co">#quarto::quarto_render(&quot;08_analysis_volcano_plot_DONE.qmd&quot;,output_format = &quot;html&quot;,execute = T)</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="co">#file.rename(from = &quot;08_analysis_volcano_plot_DONE.html&quot;,</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="co">#            to = &quot;../results/08_analysis_volcano_plot_DONE.html&quot;)</span></span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;HER2&quot;</span>)</span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>lumA <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;lumA&quot;</span>)</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>lumB <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;lumB&quot;</span>)</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>basal_like <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;bl&quot;</span>)</span></code></pre></div>
</div>
</div>
<p><strong>Load data</strong></p>
<p>In the following, the results from the differential gene expression analysis is loaded and wrangled by ise of a function <code>subset_rename_data</code> from <code>99_proj_func.R</code> . The function subsets the data to only the columns of interest for this analysis and renames the coloumns.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract only neccesary columns and rename for further joining: </span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Find only the significant proteins:</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots <span class="ot">&lt;-</span> <span class="fu">c</span>(HER2 <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_HER2 <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein),</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>                  lumA <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_lumA <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>                  lumB <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_lumB <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>                  basal_like <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_bl <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein))</span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Make a vector of the significant proteins</span></span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots_unique <span class="ot">&lt;-</span> <span class="fu">unique</span>(signif_fdr_prots)</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Select eah protein inly once.  </span></span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(signif_fdr_prots)</span></code></pre></div>
</div>
</div>
<p>Combine the datasets to the dataframe we will use:</p>
<div class="cell">
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> <span class="fu">left_join</span>(HER2, lumA, <span class="at">by =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Combine results dataframes</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(lumB, <span class="at">by =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Combine results dataframes</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(basal_like, <span class="at">by =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Combine results dataframes</span></span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(protein <span class="sc">%in%</span> signif_fdr_prots_unique)</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Filter for only the significant protiens</span></span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(basal_like, lumA, lumB, HER2, signif_fdr_prots_unique)</span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Remove unneccesary stuff from enviroenment</span></span></code></pre></div>
</div>
</div>
<p>Next, the heatmaps of the results are made using the <code>heat_map_plot</code> function from the plots file. Because we had so many significant proteins, a visualization of them all in the plot was un-informational. Therefore, we chose to only select the top 20 most significant proteins for each subgroup to display. In summary, this function…</p>
<ul>
<li><p>Takes the combined data and order it in order of significance for subtype of choice.</p></li>
<li><p>Structures the data, making sure, the proteins are plotted in order of significance of subtype in question.</p></li>
<li><p>Makes a heatmap colouring for estimate and indicating significance by the use of asterisks.</p></li>
</ul>
<div class="cell">
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>heatmap_bl <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_bl, <span class="at">title =</span> <span class="st">&quot;Basal like&quot;</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>heatmap_HER2 <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_HER2, <span class="at">title =</span> <span class="st">&quot;HER2&quot;</span>)</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>heatmap_lumA <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_lumA, <span class="at">title =</span> <span class="st">&quot;Luminal A&quot;</span>)<span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>heatmap_lumB <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_lumB, <span class="at">title =</span> <span class="st">&quot;Luminal B&quot;</span>)<span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Combine all heatmaps using &#39;patchwork&#39;</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>heatmap <span class="ot">&lt;-</span> (heatmap_bl <span class="sc">+</span> heatmap_HER2)<span class="sc">/</span>(heatmap_lumA <span class="sc">+</span>heatmap_lumB)<span class="sc">+</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&#39;collect&#39;</span>) <span class="sc">+</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;Vizualisation of results from differential gene expression analysis&quot;</span>, </span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">&quot;Using top 20 most significant proteins for each PAM50 classification&quot;</span>)</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/heatmap_combined.png&quot;</span>, heatmap, <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Save</span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb8" class="sourceCode cell-code">
<div class="sourceCode" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="co">#read files.</span></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>lumA <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>lumB <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>basal_like <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
</div>
<p>To see dispersion of differentially expressed genes, we visualize then through a volcanoplot. to make the volcanoplot, we call the function <code>volcano_plot()</code> from the <code>99_proj_func.R</code> file.</p>
<div class="cell">
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> HER2, <span class="at">subtype =</span> <span class="st">&quot;HER2&quot;</span>)</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> lumA, <span class="at">subtype =</span> <span class="st">&quot;Luminal A&quot;</span>)</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> lumB, <span class="at">subtype =</span> <span class="st">&quot;Luminal A&quot;</span>)</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">volcano_plot</span>(<span class="at">data =</span> basal_like, <span class="at">subtype =</span> <span class="st">&quot;Basal like&quot;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb12" class="sourceCode cell-code">
<div class="sourceCode" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co">#loads patchwork library to combine the four volcanoplots where / creates to row #and two columns, and collect ensures that legends are shared.</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="co">#the file with combined plot is saved as png file in &quot;/results/ directory with specified dimentions.</span></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>volcano_plots <span class="ot">&lt;-</span> (plot1 <span class="sc">+</span> plot2)<span class="sc">/</span>(plot3 <span class="sc">+</span> plot4) <span class="sc">+</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>)</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/volcano_plots.png&quot;</span>, volcano_plots, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">width =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 14 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 141 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
</div>
<div class="cell">
<div id="cb15" class="sourceCode cell-code">
<div class="sourceCode" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<p>We want to see if we get a better clustering when limiting the hierachial analysis to only consider the genes that were significant in our differential expression analysis.</p>
<p>A count matrix with a reduced dataset - with the proteomics data of only the significant proteins - were loaded.</p>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggdendro&quot;</span>)</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>cm_sig <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/count_matrix_significant_proteins.tsv&quot;</span>, <span class="at">header =</span> T) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&#39;id&#39;</span>)</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/clinical_data.tsv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header =</span> T)</span></code></pre></div>
</div>
</div>
<p>Next, the hierachial clustering was done exactly as above.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>ddg_plot <span class="ot">&lt;-</span> <span class="fu">hc_dendro_plot</span>(<span class="at">count_matrix =</span> cm_sig, <span class="at">data_type =</span> <span class="st">&quot;significant genes&quot;</span>) <span class="sc">+</span> </span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;pam50&quot;</span>)</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>ddg_plot </span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="10_analysis_dendro_signif_proteins_DONE_files/figure-html/hierachial%20clustering%202-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/ddg_plot_significant_genes.png&quot;</span>, ddg_plot, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<p><strong>Conclusion</strong>: After filtering the dataset on only the significant proteins based on our differential expression analysis, we see a much better agreement between the hierachial clustering of proteomics data and the clinical pam50 assignments.</p>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;HER2&quot;</span>)</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>lumA <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;lumA&quot;</span>)</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>lumB <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;lumB&quot;</span>)</span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>basal_like <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_rename_data</span>(<span class="at">subtype =</span> <span class="st">&quot;bl&quot;</span>)</span></code></pre></div>
</div>
</div>
<p><strong>Load data</strong></p>
<p>In the following, the results from the differential gene expression analysis is loaded and wrangled by ise of a function <code>subset_rename_data</code> from <code>99_proj_func.R</code> . The function subsets the data to only the columns of interest for this analysis and renames the coloumns.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract only neccesary columns and rename for further joining: </span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Find only the significant proteins:</span></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots <span class="ot">&lt;-</span> <span class="fu">c</span>(HER2 <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_HER2 <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein),</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>                  lumA <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_lumA <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>                  lumB <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_lumB <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>                  basal_like <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjustedfdr_bl <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein))</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Make a vector of the significant proteins</span></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots_unique <span class="ot">&lt;-</span> <span class="fu">unique</span>(signif_fdr_prots)</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Select eah protein inly once.  </span></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(signif_fdr_prots)</span></code></pre></div>
</div>
</div>
<p>Combine the datasets to the dataframe we will use:</p>
<div class="cell">
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> <span class="fu">left_join</span>(HER2, lumA, <span class="at">by =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Combine results dataframes</span></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(lumB, <span class="at">by =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Combine results dataframes</span></span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(basal_like, <span class="at">by =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>      <span class="co">#Combine results dataframes</span></span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(protein <span class="sc">%in%</span> signif_fdr_prots_unique)</span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Filter for only the significant protiens</span></span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(basal_like, lumA, lumB, HER2, signif_fdr_prots_unique)</span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Remove unneccesary stuff from enviroenment</span></span></code></pre></div>
</div>
</div>
<p>Next, the heatmaps of the results are made using the <code>heat_map_plot</code> function from the plots file. Because we had so many significant proteins, a visualization of them all in the plot was un-informational. Therefore, we chose to only select the top 20 most significant proteins for each subgroup to display. In summary, this function…</p>
<ul>
<li><p>Takes the combined data and order it in order of significance for subtype of choice.</p></li>
<li><p>Structures the data, making sure, the proteins are plotted in order of significance of subtype in question.</p></li>
<li><p>Makes a heatmap colouring for estimate and indicating significance by the use of asterisks.</p></li>
</ul>
<div class="cell">
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>heatmap_bl <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_bl, <span class="at">title =</span> <span class="st">&quot;Basal like&quot;</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>heatmap_HER2 <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_HER2, <span class="at">title =</span> <span class="st">&quot;HER2&quot;</span>)</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>heatmap_lumA <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_lumA, <span class="at">title =</span> <span class="st">&quot;Luminal A&quot;</span>)<span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>heatmap_lumB <span class="ot">&lt;-</span> <span class="fu">heat_map_results</span>(<span class="at">data =</span> combined, <span class="at">fdr_col =</span> adjustedfdr_lumB, <span class="at">title =</span> <span class="st">&quot;Luminal B&quot;</span>)<span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Combine all heatmaps using &#39;patchwork&#39;</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>heatmap <span class="ot">&lt;-</span> (heatmap_bl <span class="sc">+</span> heatmap_HER2)<span class="sc">/</span>(heatmap_lumA <span class="sc">+</span>heatmap_lumB)<span class="sc">+</span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&#39;collect&#39;</span>) <span class="sc">+</span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;Vizualisation of results from differential gene expression analysis&quot;</span>, </span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">&quot;Using top 20 most significant proteins for each PAM50 classification&quot;</span>)</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/heatmap_combined.png&quot;</span>, heatmap, <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Save</span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb8" class="sourceCode cell-code">
<div class="sourceCode" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<p>We want to see if we get a better clustering when limiting the hierachial analysis to only consider the genes that were significant in our differential expression analysis.</p>
<p>A count matrix with a reduced dataset - with the proteomics data of only the significant proteins - were loaded.</p>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggdendro&quot;</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="at">file =</span> <span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>Basal_like <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="at">file =</span> <span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>Luminal_A <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="at">file =</span> <span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>Luminal_B <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="at">file =</span> <span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>) </span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  id = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
<div id="cb13" class="sourceCode cell-code">
<div class="sourceCode" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>cm_sig <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/count_matrix_significant_proteins.tsv&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&#39;id&#39;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  id = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
</div>
<div id="cb15" class="sourceCode cell-code">
<div class="sourceCode" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/clinical_data.tsv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header =</span> T)</span></code></pre></div>
</div>
</div>
<p>Next, the hierachial clustering was done exactly as above.</p>
<div class="cell">
<div id="cb16" class="sourceCode cell-code">
<div class="sourceCode" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>ddg_plot <span class="ot">&lt;-</span> <span class="fu">hc_dendro_plot</span>(<span class="at">count_matrix =</span> cm_sig, <span class="at">data_type =</span> <span class="st">&quot;significant genes&quot;</span>) <span class="sc">+</span> </span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;pam50&quot;</span>)</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a>ddg_plot </span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="11_analysis_dendro_signif_proteins_DONE_files/figure-html/hierachial%20clustering%202-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb17" class="sourceCode cell-code">
<div class="sourceCode" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/ddg_plot_significant_genes.png&quot;</span>, ddg_plot, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<p><strong>Conclusion</strong>: After filtering the dataset on only the significant proteins based on our differential expression analysis, we see a much better agreement between the hierachial clustering of proteomics data and the clinical pam50 assignments.</p>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">

</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<p>Here, we want to see if we can determine the PAM50 groups using only the significant proteins from the linear models.</p>
<p>First, we make a PCA using the</p>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing HER2 condition based on count matrix</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>lumA <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing luminal A condition based on count matrix</span></span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a>lumB <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing Luminal B condition based on count matrix</span></span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a>basal_like <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co">#Loading results from differential expression analysis - predicing Basal like condition based on count matrix</span></span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-7"><a href="#cb138-7" aria-hidden="true" tabindex="-1"></a>protein_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="co"># loading the count matrix</span></span>
<span id="cb138-8"><a href="#cb138-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-9"><a href="#cb138-9" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots <span class="ot">&lt;-</span> <span class="fu">c</span>(HER2 <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein),</span>
<span id="cb138-10"><a href="#cb138-10" aria-hidden="true" tabindex="-1"></a>                  lumA <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb138-11"><a href="#cb138-11" aria-hidden="true" tabindex="-1"></a>                  lumB <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein), </span>
<span id="cb138-12"><a href="#cb138-12" aria-hidden="true" tabindex="-1"></a>                  basal_like <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pull</span>(protein)) <span class="co"># Extracting only significant hits from differential expression analysis. Significant hits are defined at genes that have an adjusted P-value under 0.05. </span></span>
<span id="cb138-13"><a href="#cb138-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-14"><a href="#cb138-14" aria-hidden="true" tabindex="-1"></a>signif_fdr_prots_unique <span class="ot">&lt;-</span> <span class="fu">unique</span>(signif_fdr_prots) <span class="co"># extracting only unique gene names from significant hits </span></span>
<span id="cb138-15"><a href="#cb138-15" aria-hidden="true" tabindex="-1"></a><span class="co">#length(signif_fdr_prots_unique)</span></span></code></pre></div>
</div>
</div>
<p>We can now isolate only significant hits form our count matrix.</p>
<div class="cell">
<div id="cb4" class="sourceCode cell-code">
<div class="sourceCode" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>count_matrix_reduced <span class="ot">&lt;-</span> protein_data <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(id <span class="sc">%in%</span> signif_fdr_prots_unique) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Filtering so only differential expressed genes are present. </span></span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>id, <span class="at">names_to =</span> <span class="st">&quot;id2&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Transforming to pivot_longer</span></span>
<span id="cb139-4"><a href="#cb139-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> id, <span class="at">values_from =</span> value) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># transforming it again to pivot_wider. to have the correct structure - gene names as columns and samples as rows</span></span>
<span id="cb139-5"><a href="#cb139-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;id2&quot;</span>)</span>
<span id="cb139-6"><a href="#cb139-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-7"><a href="#cb139-7" aria-hidden="true" tabindex="-1"></a>count_matrix_reduced <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;id&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">write_tsv</span>(<span class="st">&quot;../data/count_matrix_significant_proteins.tsv&quot;</span>) <span class="co"># saving results as a tsv file. </span></span></code></pre></div>
</div>
</div>
<p>PCA analysis of count matrix with only differentially expressed genes.</p>
<div class="cell">
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> count_matrix_reduced <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prcomp</span>(<span class="at">center =</span> <span class="cn">TRUE</span>,</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="cn">TRUE</span>) <span class="co"># prcomp is a function in R to transform the data into principal components. </span></span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-5"><a href="#cb140-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Scree</span></span>
<span id="cb140-6"><a href="#cb140-6" aria-hidden="true" tabindex="-1"></a>pca  <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb140-7"><a href="#cb140-7" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="st">&quot;pcs&quot;</span>)<span class="sc">|</span><span class="er">&gt;</span>  <span class="co"># From broom, we&#39;re using the tidy function to extract the nested list &quot;pcs&quot;</span></span>
<span id="cb140-8"><a href="#cb140-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> percent <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">|</span><span class="er">&gt;</span>  <span class="co"># Overriding the column Percent by multiplying current value with 100</span></span>
<span id="cb140-9"><a href="#cb140-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, </span>
<span id="cb140-10"><a href="#cb140-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> percent)) <span class="sc">+</span></span>
<span id="cb140-11"><a href="#cb140-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb140-12"><a href="#cb140-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="co"># geom_Col function creates a barplot from the global values X as the princial compoennts and &quot;Y&quot; as the results form Principal components.</span></span>
<span id="cb140-13"><a href="#cb140-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> <span class="co"># base_size = font size of the text </span></span>
<span id="cb140-14"><a href="#cb140-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Scree Plot of PCA of significant proteins&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="11_analysis_kmeans_DONE_files/figure-html/analysis%202-1.png" class="img-fluid" width="672" /></p>
</div>
</div>
<div class="cell">
<div id="cb6" class="sourceCode cell-code">
<div class="sourceCode" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a>pca_plot_axes_labels <span class="ot">&lt;-</span> pca <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="st">&quot;eigenvalues&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># extracting tidy function from broom package</span></span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lbl =</span> <span class="fu">str_c</span>(<span class="st">&quot;PC&quot;</span>, PC, <span class="st">&quot;, VE = &quot;</span>, <span class="fu">round</span>(percent<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>), <span class="st">&quot;%&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Creating a new variable called lbl where values are created by concatenating the principal compoennts and its respective variance into a character string</span></span>
<span id="cb141-4"><a href="#cb141-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="st">&quot;lbl&quot;</span>) <span class="co"># pulling the lbl value out</span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;../data/clinical_data.tsv&quot;</span>) <span class="co"># Loading the clinical data.</span></span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 77 Columns: 30
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (17): id, gender, er_status, pr_status, her2_final_status, tumor, tumor-...
dbl (13): age_at_initial_pathologic_diagnosis, days_to_date_of_last_contact,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>pca_aug <span class="ot">&lt;-</span> pca <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">augment</span>(count_matrix_reduced) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># Using the augment function from broom to extend the pca dataset with our count matrix of differentially expressed genes</span></span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">id =</span> <span class="st">&quot;.rownames&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># renaming the .rownames column to id</span></span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(clinical_data <span class="sc">|</span><span class="er">&gt;</span>  <span class="fu">select</span>(<span class="fu">c</span>(id, pam50)), <span class="co"># appending the pam50 columns from the clinical dataset to the pca dataset, based on id. </span></span>
<span id="cb144-5"><a href="#cb144-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">&quot;id&quot;</span>) </span></code></pre></div>
</div>
</div>
<p>Now, we run the k-means algorithm using n = 4, as we have 4 groups we want to determine:</p>
<div class="cell">
<div id="cb10" class="sourceCode cell-code">
<div class="sourceCode" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>k_means_clusters <span class="ot">&lt;-</span> pca_aug <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(.fittedPC1, .fittedPC2)) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># selecting PC1 and PC2</span></span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kmeans</span>(<span class="at">centers =</span> <span class="dv">4</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># initializing Kmeans clustering with 4 clusters</span></span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pluck</span>(<span class="st">&quot;cluster&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> <span class="co"># extracting &quot;cluster&quot; list.</span></span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.factor</span>() <span class="co"># converting the results to a factor for coloring and filling of PCA plot. </span></span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>PCA_kmeans <span class="ot">&lt;-</span> pca_aug <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> k_means_clusters) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> .fittedPC1, <span class="co"># PC1 on x axis</span></span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> .fittedPC2, <span class="co"># PC2 on y axis</span></span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="fu">factor</span>(k_means_clusters), <span class="co"># color based on pam50 </span></span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> <span class="fu">factor</span>(k_means_clusters))) <span class="sc">+</span> <span class="co"># fill based on pam50</span></span>
<span id="cb146-7"><a href="#cb146-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>) <span class="sc">+</span><span class="co"># creating a vertical line that interceps at x = 0</span></span>
<span id="cb146-8"><a href="#cb146-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span><span class="co"># creating a horizontal line that intercepts at y = 0</span></span>
<span id="cb146-9"><a href="#cb146-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb146-10"><a href="#cb146-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">6</span>, </span>
<span id="cb146-11"><a href="#cb146-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="co"># transparency of the dots is set to 0.5</span></span>
<span id="cb146-12"><a href="#cb146-12" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">shape =</span> pam50)) <span class="sc">+</span></span>
<span id="cb146-13"><a href="#cb146-13" aria-hidden="true" tabindex="-1"></a>          <span class="co"># global are already given above and we not just apply design for the dots </span></span>
<span id="cb146-14"><a href="#cb146-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb146-15"><a href="#cb146-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;PCA categorized through clusters&quot;</span>, <span class="co"># title of the data</span></span>
<span id="cb146-16"><a href="#cb146-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">pluck</span>(pca_plot_axes_labels, <span class="dv">1</span>), <span class="co">#name of x axis is extracted through the first value of pca_plot_axes_labels variable</span></span>
<span id="cb146-17"><a href="#cb146-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">pluck</span>(pca_plot_axes_labels, <span class="dv">2</span>), <span class="co">#name of y axis is extracted through the second value of pca_plot_axes_labels variable</span></span>
<span id="cb146-18"><a href="#cb146-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;Cluster factor&quot;</span>,</span>
<span id="cb146-19"><a href="#cb146-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;Cluster factor&quot;</span></span>
<span id="cb146-20"><a href="#cb146-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb146-21"><a href="#cb146-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="co">#setting theme to bw</span></span>
<span id="cb146-22"><a href="#cb146-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">fill =</span> pam50, <span class="at">color =</span> pam50), <span class="co">#adding informtion on the actual pam50 classifications</span></span>
<span id="cb146-23"><a href="#cb146-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">type =</span> <span class="st">&quot;t&quot;</span>, </span>
<span id="cb146-24"><a href="#cb146-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">geom =</span> <span class="st">&quot;polygon&quot;</span>, </span>
<span id="cb146-25"><a href="#cb146-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.1</span>) </span>
<span id="cb146-26"><a href="#cb146-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-27"><a href="#cb146-27" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(PCA_kmeans)</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="11_analysis_kmeans_DONE_files/figure-html/save%20results-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb12" class="sourceCode cell-code">
<div class="sourceCode" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/PCA_with_clusters.png&quot;</span>, PCA_kmeans, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code></pre></div>
</div>
</div>
<div class="cell">
<div id="cb13" class="sourceCode cell-code">
<div class="sourceCode" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<h2 id="toc-title">Table of contents</h2>
<ul>
<li><a href="#one-out-of-k-encoding-for-the-4-subgroups" id="toc-one-out-of-k-encoding-for-the-4-subgroups" class="nav-link active">One-out-of-k encoding for the 4 subgroups</a></li>
<li><a href="#correlation-matrix" id="toc-correlation-matrix" class="nav-link">Correlation matrix</a>
<ul>
<li><a href="#correlation-matrix-for-sign.-protein-expression-vs.-her2-pr-er" id="toc-correlation-matrix-for-sign.-protein-expression-vs.-her2-pr-er" class="nav-link">Correlation matrix for sign. protein expression vs. HER2, PR, ER</a>
<ul>
<li><a href="#load-differential-expression-data." id="toc-load-differential-expression-data." class="nav-link">Load differential expression data.</a></li>
<li><a href="#extract-significant-proteins" id="toc-extract-significant-proteins" class="nav-link">Extract significant proteins</a></li>
<li><a href="#retrieve-names-of-significant-proteins-mapping" id="toc-retrieve-names-of-significant-proteins-mapping" class="nav-link">Retrieve names of significant proteins (Mapping)</a></li>
</ul></li>
<li><a href="#create-separate-correlation-matrices-for-the-4-pam50-classes-significant-proteins-with-the-pr-er-and-her2-status" id="toc-create-separate-correlation-matrices-for-the-4-pam50-classes-significant-proteins-with-the-pr-er-and-her2-status" class="nav-link">Create separate correlation matrices for the 4 pam50 classes’ significant proteins with the PR, ER and her2 status</a></li>
<li><a href="#from-the-full-dataset-with-all-significant-protein-expressions-and-the-clinical-data-we-extract-only-the-columns-with-er-pr-her2-and-the-her2-classs-significant-proteins." id="toc-from-the-full-dataset-with-all-significant-protein-expressions-and-the-clinical-data-we-extract-only-the-columns-with-er-pr-her2-and-the-her2-classs-significant-proteins." class="nav-link">From the full dataset with all significant protein expressions and the clinical data we extract only the columns with ER, PR, her2 and the HER2-class’s significant proteins.</a></li>
</ul></li>
</ul>
</div>
<div id="quarto-document-content" class="content" role="main">
<div class="quarto-title">

</div>
<div class="quarto-title-meta">

</div>
<div class="cell">
<div id="cb1" class="sourceCode cell-code">
<div class="sourceCode" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div id="cb3" class="sourceCode cell-code">
<div class="sourceCode" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrr)</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Attaching package: &#39;reshape&#39;

The following object is masked from &#39;package:lubridate&#39;:

    stamp

The following object is masked from &#39;package:dplyr&#39;:

    rename

The following objects are masked from &#39;package:tidyr&#39;:

    expand, smiths</code></pre>
</div>
<div id="cb5" class="sourceCode cell-code">
<div class="sourceCode" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a><span class="co">#library(rcompanion)</span></span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;99_proj_func.R&quot;</span>)</span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-7"><a href="#cb153-7" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/full_data.tsv&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb153-8"><a href="#cb153-8" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;../data/clinical_load.csv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 105 Columns: 30
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;;&quot;
chr (17): Complete TCGA ID, Gender, ER Status, PR Status, HER2 Final Status,...
dbl (13): Age at Initial Pathologic Diagnosis, Days to Date of Last Contact,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="cb7" class="sourceCode cell-code">
<div class="sourceCode" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a>HER2 <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_HER2.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb9" class="sourceCode cell-code">
<div class="sourceCode" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>Luminal_A <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Luminal_A.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb11" class="sourceCode cell-code">
<div class="sourceCode" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a>Luminal_B <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Luminal_B.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb13" class="sourceCode cell-code">
<div class="sourceCode" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>Basal_like <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;../data/diff_exp_Basal_like.tsv&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
cols(
  protein = col_character(),
  estimate = col_double(),
  std.error = col_double(),
  statistic = col_double(),
  p.value = col_double(),
  conf.low = col_double(),
  conf.high = col_double(),
  adjusted_BH = col_double(),
  adjusted_fdr = col_double()
)</code></pre>
</div>
<div id="cb15" class="sourceCode cell-code">
<div class="sourceCode" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../data/count_matrix_1_77.tsv&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>protein_IDs <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;../_raw/77_cancer_proteomes_CPTAC_itraq.csv&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;;&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
Rows: 12553 Columns: 87
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;;&quot; chr
(86): RefSeq_accession_number, gene_symbol, gene_name, AO-A12D.01TCGA, C... dbl
(1): ...1
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
</div>
</div>
<section id="one-out-of-k-encoding-for-the-4-subgroups" class="anchored level3" data-anchor-id="one-out-of-k-encoding-for-the-4-subgroups">
<h3 class="anchored" data-anchor-id="one-out-of-k-encoding-for-the-4-subgroups">One-out-of-k encoding for the 4 subgroups</h3>
<p>We will use the binary encoded pam 50 classes for correlation, so from the single column pam50 with 4 classes we create 4 columns, one for each class, where it says 1 for the observation being in the respective class and 0 for not.</p>
<div class="cell">
<div id="cb17" class="sourceCode cell-code">
<div class="sourceCode" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>corr_kendall <span class="ot">&lt;-</span> full_data <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;er_status&quot;</span>,<span class="st">&quot;pr_status&quot;</span>, <span class="st">&quot;her2_final_status&quot;</span>, <span class="st">&quot;DGEA_HER2&quot;</span>, <span class="st">&quot;DGEA_Luminal_A&quot;</span>, <span class="st">&quot;DGEA_Luminal_B&quot;</span>, <span class="st">&quot;DGEA_Basal_like&quot;</span>)) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb165-4"><a href="#cb165-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">correlate</span>(<span class="at">method =</span> <span class="st">&quot;kendall&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb165-5"><a href="#cb165-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">focus</span>(er_status, pr_status, her2_final_status)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Correlation computed with
• Method: &#39;kendall&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
</div>
</div>
</section>
<section id="correlation-matrix" class="level1">
<h1>Correlation matrix</h1>
<p>For the geom tile function in ggplot2 a long data table is needed, so ER, PR and her2 status are pivoted to the single column status.</p>
<p>Thereafter, we can create the correlation plot for only differentially expressed genes using the function <code>corr_plot()</code> from the <code>99_proj_func.R</code> file.</p>
<div class="cell">
<div id="cb19" class="sourceCode cell-code">
<div class="sourceCode" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>c<span class="ot">&lt;-</span> corr_kendall <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">subclass =</span> term) <span class="sc">|</span><span class="er">&gt;</span>  </span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">ends_with</span>(<span class="st">&quot;status&quot;</span>), </span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;status&quot;</span>, </span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;corr&quot;</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a><span class="co">#For better readability later on, the column values in subclass and status are recoded, basically just renamed, to have them in the tile plot like this.</span></span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">subclass =</span> <span class="fu">recode</span>(subclass, </span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;DGEA_HER2&quot;</span> <span class="ot">=</span> <span class="st">&quot;HER2&quot;</span>, </span>
<span id="cb167-10"><a href="#cb167-10" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;DGEA_Luminal_A&quot;</span> <span class="ot">=</span> <span class="st">&quot;Luminal A&quot;</span>,</span>
<span id="cb167-11"><a href="#cb167-11" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;DGEA_Luminal_B&quot;</span> <span class="ot">=</span> <span class="st">&quot;Luminal B&quot;</span>,</span>
<span id="cb167-12"><a href="#cb167-12" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;DGEA_Basal_like&quot;</span> <span class="ot">=</span> <span class="st">&quot;Basal-like&quot;</span>),</span>
<span id="cb167-13"><a href="#cb167-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">status =</span> <span class="fu">recode</span>(status, </span>
<span id="cb167-14"><a href="#cb167-14" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;er_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;Estrogen&quot;</span>, </span>
<span id="cb167-15"><a href="#cb167-15" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;pr_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;Progesteron&quot;</span>,</span>
<span id="cb167-16"><a href="#cb167-16" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;her2_final_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;her2&quot;</span>),</span>
<span id="cb167-17"><a href="#cb167-17" aria-hidden="true" tabindex="-1"></a><span class="co">#The columns subclass and status are converted to factors, with levels being the unique values of the columns in the order they appear in.</span></span>
<span id="cb167-18"><a href="#cb167-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">subclass =</span> <span class="fu">factor</span>(subclass, <span class="at">levels =</span> <span class="fu">unique</span>(subclass)),</span>
<span id="cb167-19"><a href="#cb167-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">status =</span> <span class="fu">factor</span>(status, <span class="at">levels =</span> <span class="fu">unique</span>(status))</span>
<span id="cb167-20"><a href="#cb167-20" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb167-21"><a href="#cb167-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-22"><a href="#cb167-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-23"><a href="#cb167-23" aria-hidden="true" tabindex="-1"></a>res_corr_subclasses <span class="ot">&lt;-</span> <span class="fu">Corr_subclasses</span>(c)</span>
<span id="cb167-24"><a href="#cb167-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-25"><a href="#cb167-25" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/Corr_subclasses.png&quot;</span>, res_corr_subclasses , <span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<section id="correlation-matrix-for-sign.-protein-expression-vs.-her2-pr-er" class="anchored level2" data-anchor-id="correlation-matrix-for-sign.-protein-expression-vs.-her2-pr-er">
<h2 class="anchored" data-anchor-id="correlation-matrix-for-sign.-protein-expression-vs.-her2-pr-er">Correlation matrix for sign. protein expression vs. HER2, PR, ER</h2>
<section id="load-differential-expression-data." class="anchored level3" data-anchor-id="load-differential-expression-data.">
<h3 class="anchored" data-anchor-id="load-differential-expression-data.">Load differential expression data.</h3>
<p>Until now only the clinical data has been analyzed in a correlation matrix. Now it would be interesting to see, what the proteins that came out as significantly differently expressed in the four pam50-classes, are usually associated with. What do they do? Are they associated with the occurrences of proestrogen or estrogen receptors or HER2? This will only show us a correlation and no causality, however it may give an interesting outlook and ideas for further exploration.</p>
<div class="cell">
<div id="cb20" class="sourceCode cell-code">
<div class="sourceCode" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="co">#The proteins are now displayed as rows of the first column, but we need the proteins to be in columns, so that we can join the two datasets by the sample ID, so the protein names are first pivoted to a longer format and the IDs are then pivoted into a wider format. The ID column has automatically been named variable, and is now renamed id again.</span></span>
<span id="cb168-2"><a href="#cb168-2" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>id, <span class="at">names_to =</span> <span class="st">&#39;variable&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;value&#39;</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb168-3"><a href="#cb168-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> id, <span class="at">values_from =</span> value) <span class="sc">|</span><span class="er">&gt;</span>   </span>
<span id="cb168-4"><a href="#cb168-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">id =</span> variable) <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb168-5"><a href="#cb168-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(full_data <span class="sc">|</span><span class="er">&gt;</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">starts_with</span>(<span class="st">&quot;NP&quot;</span>)), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;id&quot;</span>))</span></code></pre></div>
</div>
</div>
</section>
<section id="extract-significant-proteins" class="anchored level3" data-anchor-id="extract-significant-proteins">
<h3 class="anchored" data-anchor-id="extract-significant-proteins">Extract significant proteins</h3>
<p>Now we have the merged datasets with protein expressions and subclass as well as PR/ER and her2 status data for every sample, however we will only look at the proteins that were significant in the differential expression analysis. To obtain these proteins as character vectors, the expression analysis dataset is filtered for only the significant (p-value equal to or lower than 0.05) values, then the first column is pulled, which are the protein IDs.</p>
<div class="cell">
<div id="cb21" class="sourceCode cell-code">
<div class="sourceCode" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>HER2_sign <span class="ot">&lt;-</span> HER2 <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb169-3"><a href="#cb169-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(protein)</span>
<span id="cb169-4"><a href="#cb169-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-5"><a href="#cb169-5" aria-hidden="true" tabindex="-1"></a>Basal_like_sign <span class="ot">&lt;-</span> Basal_like <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb169-6"><a href="#cb169-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb169-7"><a href="#cb169-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(protein)</span>
<span id="cb169-8"><a href="#cb169-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-9"><a href="#cb169-9" aria-hidden="true" tabindex="-1"></a>Luminal_A_sign <span class="ot">&lt;-</span> Luminal_A <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb169-10"><a href="#cb169-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb169-11"><a href="#cb169-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(protein)</span>
<span id="cb169-12"><a href="#cb169-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-13"><a href="#cb169-13" aria-hidden="true" tabindex="-1"></a>Luminal_B_sign <span class="ot">&lt;-</span> Luminal_B <span class="sc">|</span><span class="er">&gt;</span></span>
<span id="cb169-14"><a href="#cb169-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adjusted_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb169-15"><a href="#cb169-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(protein)</span>
<span id="cb169-16"><a href="#cb169-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-17"><a href="#cb169-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Sign_proteins &lt;- reduce(list(HER2_sign, Basal_like_sign, Luminal_A_sign, Luminal_B_sign), union)</span></span>
<span id="cb169-18"><a href="#cb169-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-19"><a href="#cb169-19" aria-hidden="true" tabindex="-1"></a><span class="co">#length(Sign_proteins)</span></span></code></pre></div>
</div>
</div>
</section>
<section id="retrieve-names-of-significant-proteins-mapping" class="anchored level3" data-anchor-id="retrieve-names-of-significant-proteins-mapping">
<h3 class="anchored" data-anchor-id="retrieve-names-of-significant-proteins-mapping">Retrieve names of significant proteins (Mapping)</h3>
<p>We can also obtain a list of the protein names by transforming the protein ID vector into a column of a dataframe and joining it with the ID to gene dictionary by the RefSeq_accession_number (which is for some reason called as the dataframe name, and has to be stated to be equal to the RefSeq number). This knowledge can be used later to interpret the correlations of the proteins.</p>
<div class="cell">
<div id="cb22" class="sourceCode cell-code">
<div class="sourceCode" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a>protein_mapping <span class="ot">&lt;-</span> protein_IDs <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-2"><a href="#cb170-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(RefSeq_accession_number, gene_symbol, gene_name) <span class="co"># gene_symbol &amp; gene_name has been changed from Protein.names, Gene.Names because my dataset </span></span>
<span id="cb170-3"><a href="#cb170-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-4"><a href="#cb170-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-5"><a href="#cb170-5" aria-hidden="true" tabindex="-1"></a>HER2_mapping <span class="ot">&lt;-</span> HER2_sign <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-6"><a href="#cb170-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-7"><a href="#cb170-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_mapping, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;HER2_sign&quot;</span> <span class="ot">=</span> <span class="st">&quot;RefSeq_accession_number&quot;</span>))</span>
<span id="cb170-8"><a href="#cb170-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-9"><a href="#cb170-9" aria-hidden="true" tabindex="-1"></a>Basal_like_mapping <span class="ot">&lt;-</span> Basal_like_sign <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-10"><a href="#cb170-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-11"><a href="#cb170-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_mapping, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;Basal_like_sign&quot;</span> <span class="ot">=</span> <span class="st">&quot;RefSeq_accession_number&quot;</span>))</span>
<span id="cb170-12"><a href="#cb170-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-13"><a href="#cb170-13" aria-hidden="true" tabindex="-1"></a>Luminal_A_mapping <span class="ot">&lt;-</span> Luminal_A_sign <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-14"><a href="#cb170-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-15"><a href="#cb170-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_mapping, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;Luminal_A_sign&quot;</span> <span class="ot">=</span> <span class="st">&quot;RefSeq_accession_number&quot;</span>))</span>
<span id="cb170-16"><a href="#cb170-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-17"><a href="#cb170-17" aria-hidden="true" tabindex="-1"></a>Luminal_B_mapping <span class="ot">&lt;-</span> Luminal_B_sign <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-18"><a href="#cb170-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|</span><span class="er">&gt;</span> </span>
<span id="cb170-19"><a href="#cb170-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_mapping, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;Luminal_B_sign&quot;</span> <span class="ot">=</span> <span class="st">&quot;RefSeq_accession_number&quot;</span>))</span></code></pre></div>
</div>
</div>
</section>
</section>
<section id="create-separate-correlation-matrices-for-the-4-pam50-classes-significant-proteins-with-the-pr-er-and-her2-status" class="anchored level2" data-anchor-id="create-separate-correlation-matrices-for-the-4-pam50-classes-significant-proteins-with-the-pr-er-and-her2-status">
<h2 class="anchored" data-anchor-id="create-separate-correlation-matrices-for-the-4-pam50-classes-significant-proteins-with-the-pr-er-and-her2-status">Create separate correlation matrices for the 4 pam50 classes’ significant proteins with the PR, ER and her2 status</h2>
</section>
<section id="from-the-full-dataset-with-all-significant-protein-expressions-and-the-clinical-data-we-extract-only-the-columns-with-er-pr-her2-and-the-her2-classs-significant-proteins." class="anchored level2" data-anchor-id="from-the-full-dataset-with-all-significant-protein-expressions-and-the-clinical-data-we-extract-only-the-columns-with-er-pr-her2-and-the-her2-classs-significant-proteins.">
<h2 class="anchored" data-anchor-id="from-the-full-dataset-with-all-significant-protein-expressions-and-the-clinical-data-we-extract-only-the-columns-with-er-pr-her2-and-the-her2-classs-significant-proteins.">From the full dataset with all significant protein expressions and the clinical data we extract only the columns with ER, PR, her2 and the HER2-class’s significant proteins.</h2>
<p>Thereafter, we can create the correlation plot for only differentially expressed genes using the function <code>corr_plot()</code> from the <code>99_proj_func.R</code> file.</p>
<div class="cell">
<div id="cb23" class="sourceCode cell-code">
<div class="sourceCode" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To reduce redundancy, we will call the results correlation plot from the function file. The second input is all the genes that are significant for the respective test. </span></span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-3"><a href="#cb171-3" aria-hidden="true" tabindex="-1"></a>Her2_plot <span class="ot">&lt;-</span> <span class="fu">corr_plot</span>(cts,HER2_sign) <span class="sc">+</span> </span>
<span id="cb171-4"><a href="#cb171-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;HER2 positive&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Status&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Protein&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Correlation&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using an external vector in selections was deprecated in tidyselect 1.1.0.
ℹ Please use `all_of()` or `any_of()` instead.
  # Was:
  data %&gt;% select(test)

  # Now:
  data %&gt;% select(all_of(test))

See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Non-numeric variables removed from input: `id`
Correlation computed with
• Method: &#39;kendall&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
</div>
<div id="cb26" class="sourceCode cell-code">
<div class="sourceCode" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a>Luminal_A_plot <span class="ot">&lt;-</span> <span class="fu">corr_plot</span>(cts, <span class="fu">head</span>(Luminal_A_sign,<span class="dv">5</span>)) <span class="sc">+</span> </span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Luminal A&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Status&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Protein&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Correlation&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Non-numeric variables removed from input: `id`
Correlation computed with
• Method: &#39;kendall&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
</div>
<div id="cb28" class="sourceCode cell-code">
<div class="sourceCode" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a>Luminal_B_plot <span class="ot">&lt;-</span> <span class="fu">corr_plot</span>(cts, Luminal_B_sign) <span class="sc">+</span> </span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Luminal B&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Status&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Protein&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Correlation&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Non-numeric variables removed from input: `id`
Correlation computed with
• Method: &#39;kendall&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
</div>
<div id="cb30" class="sourceCode cell-code">
<div class="sourceCode" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>Basal_like_plot <span class="ot">&lt;-</span> <span class="fu">corr_plot</span>(cts, <span class="fu">head</span>(Basal_like_sign,<span class="dv">5</span>)) <span class="sc">+</span> </span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Basal Like&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Status&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Protein&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Correlation&quot;</span>)</span></code></pre></div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Non-numeric variables removed from input: `id`
Correlation computed with
• Method: &#39;kendall&#39;
• Missing treated using: &#39;pairwise.complete.obs&#39;</code></pre>
</div>
<div id="cb32" class="sourceCode cell-code">
<div class="sourceCode" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span>(Basal_like_plot <span class="sc">+</span> Her2_plot) <span class="sc">/</span> (Luminal_A_plot <span class="sc">+</span> Luminal_B_plot)<span class="sc">+</span></span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&#39;collect&#39;</span>) <span class="sc">+</span></span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;Correlation of PR, ER and HER2 status with significant proteins&quot;</span>, </span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">&quot;Using top most significant proteins for each PAM50 classification&quot;</span>)<span class="sc">&amp;</span></span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>), <span class="co"># Adjust title size and horizontal position</span></span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>) <span class="co"># Adjust subtitle size and horizontal position</span></span>
<span id="cb180-8"><a href="#cb180-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb180-9"><a href="#cb180-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-10"><a href="#cb180-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(p4)</span></code></pre></div>
</div>
<div class="cell-output-display">
<p><img src="12_correlation_DONE_files/figure-html/results%20and%20save%20plots-1.png" class="img-fluid" width="672" /></p>
</div>
<div id="cb33" class="sourceCode cell-code">
<div class="sourceCode" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;../results/Protein_corr_4p.png&quot;</span>, p4 , <span class="at">height =</span> <span class="dv">8</span>, <span class="at">width =</span> <span class="dv">10</span>)</span></code></pre></div>
</div>
</div>
<pre><code>::: {.cell}

```{.r .cell-code}
rm(list = ls())</code></pre>
<p>:::</p>
</section>
</section>
</div>
</div>
</body>
</html>
