---
title: "Cluster analysis with all proteomic data"
format: html
editor: visual
---

### Analysis based on significant proteins

We want to see if we get a better clustering when limiting the hierachial analysis to only consider the genes that were significant in our differential expression analysis.

A count matrix with a reduced dataset - with the proteomics data of only the significant proteins - were loaded.

```{r}
#| label: load data
library(tidyverse)
source("99_proj_func.R")
library("ggdendro")

cm_sig <- read.table("../data/count_matrix_significant_proteins.tsv", header = T) |>
  column_to_rownames('id')
clinical_data <- read.table("../data/clinical_data.tsv", sep = "\t", header = T)
```

Next, the hierachial clustering was done exactly as above.

```{r}
#| label: hierachial clustering 2
ddg_plot <- hc_dendro_plot(count_matrix = cm_sig, data_type = "significant genes") + 
  labs(color = "pam50")
ddg_plot 

ggsave("../results/ddg_plot_significant_genes.png", ddg_plot, height = 5, width = 6)
```

**Conclusion**: After filtering the dataset on only the significant proteins based on our differential expression analysis, we see a much better agreement between the hierachial clustering of proteomics data and the clinical pam50 assignments.
